Objects/dictobject.c

```diff
@@ -2511,6 +2511,40 @@ _PyDict_LoadGlobalStackRef(PyDictObject *globals, PyDictObject *builtins, PyObje
     assert(ix >= 0 || PyStackRef_IsNull(*res));
 }
 
+PyObject *
+_PyDict_LoadBuiltinsFromGlobals(PyObject *globals)
+{
+    if (!PyDict_Check(globals)) {
```

Comment:
maybe just add assertion as it is internal only API

Reply:
Some of the callers can pass in a non-dict "globals". IIRC, `test_eval_code_ex.py` tests for this case and expects a `SystemError`.